python/lsst/ap/association/diaCalculationPlugins.py

```diff
@@ -0,0 +1,158 @@
+        else:
+            diaObject["ra"] = aveCoord.getRa().asDegrees()
+            diaObject["decl"] = aveCoord.getDec().asDegrees()
+
+    def fail(self, diaObject, error=None):
+        """Set diaObject position values to nan.
+
+        Parameters
+        ----------
+        diaObject : `dict`
+            Summary object to store values in.
+        error : `BaseException`
+            Error to pass.
+        """
+        diaObject["ra"] = np.nan
+        diaObject["decl"] = np.nan
+
+
+class WeightedMeanDiaPsFluxConfig(DiaObjectCalculationPluginConfig):
+    pass
+
+
+@register("ap_meanFlux")
+class WeightedMeanDiaPsFlux(DiaObjectCalculationPlugin):
+    """Compute the weighted mean and mean error on the point source fluxes
+    of the DiaSource measured on the difference image.
+    """
+
+    ConfigClass = WeightedMeanDiaPsFluxConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_CATALOGCALCULATION
+
+    def __init__(self, config, name, metadata):
+        DiaObjectCalculationPlugin.__init__(self, config, name, metadata)
+
+    def calculate(self,
+                  diaObject,
+                  diaSources,
+                  filterDiaFluxes,
+                  filterName,
+                  **kwargs):
+        """Compute the mean ra/dec position of the diaObject given the
+        diaSource locations.
+
+        Parameters
+        ----------
+        diaObject : `dict`
+            Summary object to store values in.
+        diaSources : `pandas.DataFrame`
+            DataFrame representing all diaSources associated with this
+            diaObject.
+        filterDiaFluxes : `pandas.DataFrame`
+            DataFrame representing diaSources associated with this
+            diaObject that are observed in the band pass ``filterName``.
+        filterName : `str`
+            Simple name of the filter for the flux being calculated.
+        """
+        if len(filterDiaFluxes) > 0:
+            tot_weight = np.nansum(1 / filterDiaFluxes["psFluxErr"] ** 2)
+            fluxMean = np.nansum(filterDiaFluxes["psFlux"] /
+                                 filterDiaFluxes["psFluxErr"] ** 2)
+            fluxMean /= tot_weight
+            fluxMeanErr = np.sqrt(1 / tot_weight)
+        else:
+            fluxMean = np.nan
+            fluxMeanErr = np.nan
+
+        if np.isfinite(fluxMean) and np.isfinite(fluxMeanErr):
+            diaObject["%sPSFluxMean" % filterName] = fluxMean
+            diaObject["%sPSFluxMeanErr" % filterName] = fluxMeanErr
+        else:
+            self.fail(diaObject, filterName)
+
+    def fail(self, diaObject, filterName, error=None):
+        """Set diaObject position values to nan.
+
+        Parameters
+        ----------
+        diaObject : `dict`
+            Summary object to store values in.
+        filterName : `str`
+            Simple name of the filter for the flux being calculated.
+        error : `BaseException`
+            Error to pass.
```

Comment:
Nowhere as yet. This is a hold over from the CatalogCaculation pluggins. Since we don't have flags defined for DiaObject, I've left this as is for now.